File: src/bin/regular.rs
use std::sync::{Arc, Mutex};
use std::thread;
use std::time::Duration;

fn main() {
    let output = Arc::new(Mutex::new(Vec::new()));
    let output_clone = output.clone();
    thread::spawn(move || fill_vector(&output_clone));
    loop {
        thread::sleep(Duration::from_millis(100));
        println!("{:?}", output.lock().unwrap());
    }
}

fn fill_vector(output: &Mutex<Vec<u64>>) {
    for i in 0.. {
        thread::sleep(Duration::from_secs(1));
        output.lock().unwrap().push(i);
    }
}

File: regular.py
#! /usr/bin/env python3

import threading
import time


def main():
    output = []
    mutex = threading.Lock()
    threading.Thread(
        target=fill_vector,
        args=(output, mutex),
        daemon=True,
    ).start()
    while True:
        time.sleep(0.1)
        with mutex:
            print(output)


def fill_vector(output, mutex):
    for i in range(2**64):
        time.sleep(1)
        with mutex:
            output.append(i)


if __name__ == "__main__":
    main()
